feat(SendGrid): Add region selection for API requests and update vers…

kowshik24 opened the pull request:
## Summary

<img width="1687" height="845" alt="Image" src="https://github.com/user-attachments/assets/a499ee9e-d4b7-4a61-b254-ae79a9cd26b1" />

This PR adds configurable region selection for SendGrid API requests, allowing users to choose between US and EU endpoints. Previously, the SendGrid integration was hardcoded to use only the US API endpoint (`api.sendgrid.com`), which prevented users with European data residency requirements from using their EU SendGrid accounts.

### Changes Made:
- Added a "Region" parameter to the SendGrid node with options for US and EU
- Updated the `sendGridApiRequest` function to dynamically select the appropriate API endpoint based on the selected region:
  - US region: `api.sendgrid.com/v3`
  - EU region: `api.eu.sendgrid.com/v3`
- Bumped the SendGrid node version from 1 to 2 to reflect this breaking change

### Testing:
Users can now:
1. Select their preferred region (US or EU) in the SendGrid node
2. Use EU SendGrid accounts with the correct API endpoint
3. Continue using US SendGrid accounts as before

## Related Linear tickets, Github issues, and Community forum posts

Fixes #19657

## Review / Merge checklist

- [x] PR title and summary are descriptive.
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

---

**Additional Notes:**
- This is a breaking change since existing workflows will need to specify the region parameter
- The default region is set to 'us' to maintain backward compatibility for existing workflows
- The implementation follows the same pattern used by other nodes like Customer.io for region selection

kowshik24 committed:
feat(SendGrid): Add region selection for API requests and update version to 2

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=19816) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=19816) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=19816) it.</sub>

Joffcom commented:
Hey @kowshik24,

Thank you for your contribution. We appreciate the time and effort you’ve taken to submit this pull request.

Before we can proceed, please ensure the following:
	•	Tests are included for any new functionality, logic changes or bug fixes.
	•	The PR aligns with our contribution guidelines.

Regarding new nodes:
We no longer accept new nodes directly into the core codebase. Instead, we encourage contributors to follow our Community Node Submission Guide to publish nodes independently.

If your node integrates with an AI service that you own or represent, please email nodes@n8n.io and we will be happy to discuss the best approach.

About review timelines:
This PR has been added to our internal tracker as "GHC-4540". While we plan to review it, we are currently unable to provide an exact timeframe. Our goal is to begin reviews within a month, but this may change depending on team priorities. We will reach out when the review begins.

Thank you again for contributing to n8n.

kowshik24 committed:
feat(SendGrid): Add region selection for API requests and update version to 2

- Add region parameter with US/EU options
- Fix getNodeParameter call syntax in GenericFunctions
- Update node version to 2 for breaking change
- Update tests to include region functionality

Fixes #19657
